library/alloc/src/vec/mod.rs

```diff
@@ -1711,6 +1711,11 @@ impl<T, A: Allocator> Vec<T, A> {
         F: FnMut(&mut T) -> bool,
     {
         let original_len = self.len();
+
+        if original_len == 0 {
+            return;
+        }
```

Comment:
Could you add a comment saying this allows for better optimization? Just so it doesn't get inadvertently deleted.

Reply:
Sure, added